rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/50_synonym_rule_put.yml

```diff
@@ -0,0 +1,78 @@
+setup:
+  - skip:
+      version: " - 8.8.99"
+      reason: Introduced in 8.9.0
+  - do:
+      synonyms.put:
+        synonyms_set: test-synonyms
+        body:
+          synonyms_set:
+            - synonyms: "hello, hi"
+              id: "test-id-1"
+            - synonyms: "bye => goodbye"
+              id: "test-id-2"
+            - synonyms: "test => check"
+              id: "test-id-3"
+
+
+---
+"Update a synonyms rule":
+  - do:
+      synonym_rule.put:
+        synonyms_set: "test-synonyms"
+        synonym_rule: "test-id-2"
+        body:
+          synonyms: "bye, goodbye, seeya"
+
+  - match: { result: "updated" }
+
+  - do:
+      synonyms.get:
+        synonyms_set: test-synonyms
+
+  - match:
+      synonyms_set:
+        - synonyms: "hello, hi"
+          id: "test-id-1"
+        - synonyms: "bye, goodbye, seeya"
+          id: "test-id-2"
+        - synonyms: "test => check"
+          id: "test-id-3"
+
+
+---
+"Create a new synonym rule":
```

Comment:
Possible additional test case: Create a synonym rule for a synonym set that does not yet exist and validate the expected response.

Reply:
I think that's included in line 70 😉